substrings.rb

```diff
@@ -0,0 +1,8 @@
+def substrings(string, dictionary)
+    count_hash = {}
```

Comment:
a great convention for hash naming - `[value]_by_[key]` (also applies to FE - but there it's `[value]By[Key]`).

here - the keys are the substrings and the values are the occurrences counts of those substrings in the string
so what about
```suggestion
    occurrences_count_by_substring = {}
```

why? because it gives you (and the next developer who will read your code) a great notion of what is stored in this object - and later when (sometime) you do
`occurrences_count = occurrences_count_by_substring[substring]`
or
`occurrences_count_by_substring.each do |substring, occurrences_count|`

it's perfectly clear what you do.

Reply:
okay, renamed